cfpb/cfgov-refresh
# Hypothesis #3: Migrate WP data into Django models for Events

In order to not lose data from WP, we need a mechanism to convert the data that's in the Wordpress database and put it into our Django backend. This is that mechanism. For now, it only supports migrating the `event` post type in Wordpress, but as we add more processors, we'll be able to migrate all of the data over.

The way the command `import-data` works is like this:

`import-data arg1 arg2 arg3 -u username -p password --app appname`
- `arg1` is the WP post type defined in the processors.py file. For this PR, it's `events`
- `arg2` is the Django model name where the data is going to go. For this PR, it's `eventpage`
- `arg3` is the slug of the parent page that the pages will exist under. You will create this later as `events`
- `-u` and `-p` are you credentials to your created superuser
- `--app` is the name of the app the Django models from `arg2` exist in. It defaults to our app, `v1`
- `--overwrite` overwrites existing pages in Wagtail based on comparing slugs. Be careful when using this as it will overwrite the data in Wagtail with data from the source (Wordpress). 
- `--verbosity` is set to 1 by default. Set it to higher than 1 and expect the name of the slugs to appear where appropriate.

## Additions

- Management command called `import-data`
- EventPage and EventLandingPage models for the migrating data
- Wagtail processor for Events
- Some settings changes and a migration script

## Testing

- Run `python cfgov/manage.py migrate`
- Run `./runserver`
- Go to localhost:8000/admin/, login, and under the root page create an `EventLandingPage` called "Events" and make sure that the slug is set to "events". (Click on promote tab to check)
- Run `python cfgov/manage.py migrate-wp-data events eventpage events -u <your username> -p <yourpass>`
- Go to localhost:8000/admin/ and check under the Event Landing page you made and see if the data is all there.
- Additionally, check if the overwrite flag works by making a change in one of the events, then adding `--overwrite` to the end of the command and running it again. The change should be overwritten.

## Review

- @kave 
- @rosskarchner
